### DISCUSSION TITLE
No connection without starting the app / search Spotify not working

### DISCUSSION THREAD

**Discussion by 1dilemma2** [BimmerGestalt/AAIdrive#752] [General]
Hi. I had everything set up and working fine. Then there was an update of myBMW which was buggy. So I fiddled around a lot and ended up in uninstall everything and start from scratch.

Now most things work like before except two things:

- If I start the car I have to start AAiDrive manually on my phone to get the connection. Before it started automatically.
- I can't search for custom terms (Bands or songs) anymore. It always shows "no result". Also this worked before. Now I can only browse my Spotify library but no search for specific things

Any idea what happened and what's the difference? Regarding the spotify problem I realized that the checkbox "Search for Spotify" (the third one) is not checked. I tried it several times and authorized Spotify several times but the checkbox still is unchecked.

Thank you for your help.

**Comment by hufman** [comment]
I'm glad you have enjoyed this project!
AAIdrive is meant to start up automatically with the Bluetooth connection starts up. Occasional changes to the Android permission system have made that more difficult, and it is supposed to notice and have a button to prompt for permission, but can you manually check the permissions and see if AAIdrive has Nearby Devices permission?
I'm sorry to hear about the Spotify API permission, can you try removing the API connection specifically from the Spotify Apps page? http://www.spotify.com/account/apps/
Afterwards, open the AAIdrive Music tab and tap into Spotify to open the mini player, and then test using the Search tab there.

**Reply by 1dilemma2** [reply]
Hi Walter. Yes, I was very glad to find your app. Was a bit disappointed after I realized that my car doesn't support Android Auto. But with your app...Thanks.

I checked the nearby permission and it wasn't granted. So I will check later if it works.

The Spotify thing is strange. I already removed the API connection in Spotify directly and readded it in AAiDrive again. Same behaviour. I can grant both permissions in AAiDrive but only the first one is checked after it. Although it states that the API permission is set after I grant it for the second checkbox.

Even more strange is that it worked until yesterday morning before I had to fiddle around...

**Comment by 1dilemma2** [comment]
So. The problem with the automatic connection is solved. It was the nearby permission. Thank you.

Only the Spotify problem still occures.

**Reply by 1dilemma2** [reply]
The problem with Spotify is also solved. 

Solution:
Don't delete permissions and grant them again without waiting a bit. In my case I deleted the permission on Spotify an granted them again immediatly in AAiDrive. 

Yesterday I thought that this could be the cause of the problem and waited about 1 hour after deleting the permission. After this time I could grant both permissions in AAiDrive and everything works like it should...

**Reply by hufman** [reply]
That's great, I'm glad you got it working!

**Reply by 1dilemma2** [reply]
Hi again. Don't know why but since today I can't search Spotify anymore. I didn't change anything in the meantime. I tried to delete the permissions on the Spotify homepage and grant them through AAiDrive again (after on hour). No success. Uninstalled AAiDrive and reinstalled it. Checked all permissions. No success. I can play and control Spotify over iDrive and browse my library but searching does not work....

Don't know what else I can try...

**Reply by 1dilemma2** [reply]
Tried it once more. I set the permission for library. The Spotify site opens and I can grant the permission. Then I try to set the permission for searching Spotify. The Spotify site opens again and I grant this permission but in AAiDrive the checkbox stays empty...

**Reply by hufman** [reply]
I'm unsure what to try next, unfortunately.

**Reply by 1dilemma2** [reply]
It's a bit strange. After waiting over night, it was possible to grant the rights again for Spotify search. Then all worked in the car for the next 5 times I connected the phone to the car. On the 6th connection the Spotify search function wasn't avaliable anymore. But after I left the car and came back, it worked again. I didn't change anything regarding the connection or needed apps between my phone and the car.

Now I will have an eye on it. For the moment it seems to work most of the time. I wonder how long:-).
